packages/scene-composer/src/components/SceneLayers.tsx

```diff
@@ -31,11 +36,32 @@ export const SceneLayers: React.FC = () => {
     refetchOnWindowFocus: false,
   });
 
+  const checkForSceneRootEntity = useCallback(async () => {
+    const sceneMetadataModule = getGlobalSettings().twinMakerSceneMetadataModule;
+    if (sceneMetadataModule) {
+      if (!(await checkIfEntityExists(sceneRootEntityId, sceneMetadataModule))) {
+        addMessages([
+          {
+            category: DisplayMessageCategory.Error,
+            messageText: formatMessage({
+              description: 'Scene Entity Error',
+              defaultMessage: 'Dynamic Scene is missing root scene entity',
+            }),
+          },
+        ]);
+      }
+    }
+  }, [sceneRootEntityId, addMessages, formatMessage]);
+
   useEffect(() => {
     if (nodes.data) {
-      renderSceneNodes(nodes.data);
+      if (nodes.data.length > 0) {
```

Comment:
Wouldn't you need to check if there's a scene root entity no matter what? What is the `nodes` object going to look like if `sceneRootEntityId` doesn't exist?

Reply:
nit-pick: you can just put `if (nodes.data.length)` because any length is more than 0. 👍

Reply:
So if  the scene has a sceneRootEntityID the query gets run.  So if(nodes.date) guards against static scenes where we didn't run the query and also don't need to check for the root entity.  When the scene is dynamic the query gets run and nodes.data exists but may be length zero.  If it's not length zero it means the query based around the root so we don't need to manually check for it. But if it's zero then we need to go check and see if the query came back empty because there was no root.  It could come back length zero because there is a valid scene root but nothing placed in the scene to be child of it.

Reply:
That nit pick is a logic error because length can be zero because data is an array so it's length can be present and set to zero.